tsconfig.json

```diff
@@ -0,0 +1,15 @@
+{
+  "compilerOptions": {
+    "outDir": "dist",
+    "target": "es2016",
+    "module": "commonjs",
+    "moduleResolution": "node",
+    "esModuleInterop": true,
+    "forceConsistentCasingInFileNames": true,
+    "strict": false,
```

Comment:
If you're modernising (and rewriting in TS), does it make sense to start with some stricter TS defaults? Or just doing the minimum to move it from JS for the first pass?

Reply:
minimum, because we still want to keep the long standing support astring has. 

will probably make a more major release with everything moved to esm as well